### ISSUE TITLE
expose classes

### ISSUE THREAD

**Issue by benman1** [yglukhov/nimpy#146]
Hi
I think this project is really cool, and potentially very useful. I've tried the simple examples, and it's like a breeze, very nice. However, the examples with arrays and custom types look very tedious.

I was using C and C++ for python extensions using boost, CFFI and other mechanisms. Basically, you define your C/C++ struct or class as a python class, and this is very convenient. Here's the [boost way of doing things](https://www.boost.org/doc/libs/1_68_0/libs/python/doc/html/tutorial/tutorial/exposing.html). Do you think you could implement this somehow?

Thanks!
Ben.

**Comment by benman1** [comment]
Analogously to the exportc pragma it could look like this:

object.nim:

```
import nimpy

type
  MySeq {.exportpy} = object
    len, cap: int

proc return_struct(num: int): MySeq{.exportpy.} =
  return MySeq(len: num, cap: num)

```
object.py: (test)

```
import object
my_struct = object.MyStruct()
my_struct.cap = 5
my_struct.len = 5
assert object.greet(5).num == my_struct.num
```

**Comment by benman1** [comment]
A workaround for now could be to use the emit pragma and boost. Roughly something like this:

```nim
{.emit: """
#include <boost/python.hpp>

class_<MyStruct>("MyStruct")
     .def_readwrite("len", &Num::len)
     .def_readwrite("cap", &Num::cap)
"""
.}
```

**Comment by benman1** [comment]
I've just noticed there's [experimental support for type export](https://github.com/yglukhov/nimpy/blob/91da55d7ea9f9bc2ee4aa0ecc8b703aed91d34b5/nimpy.nim#L1154). I'll try it out.

**Comment by cvanelteren** [comment]
Is there any update on this feature?

**Comment by yglukhov** [comment]
Basic support implemented in #210.
